test/common/crudtester.go

```diff
@@ -364,9 +343,15 @@ func (c *FederatedTypeCrudTester) CheckPropagation(template, placement, override
 
 func (c *FederatedTypeCrudTester) waitForResource(client clientset.Interface, qualifiedName federatedtypes.QualifiedName, expectedVersion string) error {
 	targetAdapter := c.adapter.Target()
-	err := wait.PollImmediate(c.waitInterval, c.clusterWaitTimeout, func() (bool, error) {
+	compareHelper, err := util.NewComparisonHelper(targetAdapter.VersionCompareType())
```

Comment:
I think the comparison helper should be initialized on crudtester initialization, as with the sync controller.  The error that can occur won't change, so detecting as early as possible is desirable.

Reply:
I think so too but opted for this as an expedient to get to e2e testing asap.